src/snowflake/snowpark/modin/plugin/io/snow_io.py

```diff
@@ -665,13 +655,11 @@ def read_sas(cls, **kwargs):  # noqa: PR01
         return cls.from_pandas(pandas.read_sas(**kwargs))
 
     @classmethod
-    @pandas_module_level_function_not_implemented()
-    def read_pickle(
-        cls,
-        filepath_or_buffer,
-        **kwargs,
-    ):
-        pass  # pragma: no cover
+    def read_pickle(cls, **kwargs) -> SnowflakeQueryCompiler:
```

Comment:
This one... I think I would either not implement this one, or force the user to acknowledge that they trust the underlying data they are de-pickling by forcing the use of an argument:

`pd.read_pickle(unpicking_trusted_data=False)`

Only if the user flips that to "True" would it make sense. I would run that by @sfc-gh-dolee though.